src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+                                     || EventSystem.current.currentSelectedGameObject is null
+                                     || this._currentBuyTransaction is null)
+                return;
 
-            } else if (Commands.FirePos && !_selectionManager.Empty) {
-                EnterFirePosMode();
+            closestSpawn.BuyPlatoons(this._currentBuyTransaction.GhostPlatoons);
 
-            } else if (Commands.ReverseMove && !_selectionManager.Empty) {
-                EnterReverseMoveMode();
-
-            } else if (Commands.FastMove && !_selectionManager.Empty) {
-                EnterFastMoveMode();
-            }
+            if (KeyDown(KeyCode.LeftShift))
+                this._currentBuyTransaction = this._currentBuyTransaction.Clone();
+            else
+                this.ExitPurchasingMode();
         }
 
-        private void EnterFirePosMode()
+        private void MoveGhostsToMouse()
         {
-            CurMouseMode = MouseMode.firePos;
-            Vector2 hotspot = new Vector2(_firePosReticle.width / 2, _firePosReticle.height / 2);
-            Cursor.SetCursor(_firePosReticle, hotspot, CursorMode.Auto);
+            if (Util.GetTerrainClickLocation(out RaycastHit hit))
+                this._selectionManager.PrepareMoveOrderPreview(hit.point);
         }
 
-        private void EnterNormalMode()
+        private void OnOrderHold()
         {
-            CurMouseMode = MouseMode.normal;
-            Cursor.SetCursor(null, Vector2.zero, CursorMode.Auto);
+            if (Util.GetTerrainClickLocation(out RaycastHit hit))
+                this._selectionManager.RotateMoveOrderPreview(hit.point);
         }
 
-        private void EnterFastMoveMode()
+        private void SetCursor(Modes modes)
         {
-            CurMouseMode = MouseMode.fastMove;
-            Vector2 hotspot = new Vector2(0, 0);
-            Cursor.SetCursor(_primedReticle, hotspot, CursorMode.Auto);
+            this.MouseState = modes;
+            Vector2 nextMouseLocation = Vector2.zero;
+            Texture2D nextMouseTexture = Texture2D.whiteTexture;
+            switch (modes)
+            {
+                case Modes.FirePosition:
+                    nextMouseLocation = new Vector2(this._firePosReticle.width / 2, this._firePosReticle.height / 2);
+                    break;
+                case Modes.FastMove:
+                    nextMouseTexture = this._primedReticle;
+                    break;
+                case Modes.Normal:
+                    nextMouseTexture = null;
+                    break;
+                case Modes.ReverseMove:
+                    nextMouseTexture = this._primedReticle;
+                    break;
+                case Modes.Purchasing: break;
+                default: throw new ArgumentOutOfRangeException(nameof(modes), modes, null);
+            }
+
+            Cursor.SetCursor(nextMouseTexture, nextMouseLocation, CursorMode.Auto);
```

Comment:
I find this less readable. With four methods you can have it stated black on white: `Cursor.SetCursor(_primedReticle, Vector2.zero)`. Now I have to read this switch statement that changes a different variable in each clause? What are we gaining here?

Reply:
I went with what was there and tried to modify things for *ideas* and **suggestions**. I did my best to avoid deleting entire snippets.